portal/static/js/gil.js

```diff
@@ -854,214 +955,425 @@ module.exports = OrgTool = (function() {
+              self.handleItemRedirect(userId, "symptom-tracker", found_symptom_tracker);
+            };
+            window.app.utilObj.setVis(false);
+          }).fail(function() {
+            window.app.utilObj.setVis(false);
+          });
         };
+      };
   })();
-},{}]
-},{},[2])
+},{}],
+14:[function(require,module,exports){
+var accessCodeObj;
 
-function checkBannerStatus() {
-  if (typeof sessionStorage != "undefined") {
-    var data = sessionStorage.getItem('bannerAccessed');
-    if (data == "yes") {
-      $('.js-close-upper-banner').trigger("click");
+module.exports = accessCodeObj = (function() {
+  return function() {
+    this.handleAccessCode = function() {
+      if ($("#shortcut_alias").val() !== "" && $("#access_code_error").text() === "") {
+        $("#access_code_info").show();
+        $("#accessCodeLink").addClass("icon-box__button--disabled");
+        $("#btnCreateAccount").removeAttr("href").addClass("icon-box__button--disabled");
+        window.app.interventionSessionObj.setInterventionSession();
+        setTimeout(function() { location.replace("/go/" + ($("#shortcut_alias").val()).toLowerCase());}, 4000);
+      } else {
+        if ($("#access_code_error").text() !== "") {
+          $("#access_code_error").show();
+        }
+      }
     };
-  }
-};
+    this.handleEvents = function() {
+        var self = this;
+        $("#shortcut_alias").on("keyup paste", function(e) {
+          if ($(this).val() !== "") {
+            window.app.orgTool.validateIdentifier(false,
+              function(data) {
+                if (data) {
+                  if (data.name) {
+                    $("#access_code_info").html(i18next.t("Thank you, your access code {shortcut_alias} indicates you are located at the {org_name}. Proceeding to registration ...").replace("{shortcut_alias}", "<b>"+$("#shortcut_alias").val()+"</b>").replace("{org_name}", "<b>"+data.name+"</b>"));
+                      $("#access_code_error").text("");
+
+                    } else {
+                      if (data.error) {
+                        $("#access_code_info").text("");
+                        $("#access_code_error").text(i18next.t("You have entered an invalid access code.  Please try again"));
+                      }
+                  }
+                } else {
+                    if ($("#shortcut_alias").val() !== "") {
+                      $("#access_code_info").text("");
+                      $("#access_code_error").text(i18next.t("System was unable to process your request."));
+                    }
+                }
 
-function setSelectedNavItem(obj) {
-    $(obj).addClass("side-nave-items__item--selected");
-    $("li.side-nave-items__item--selected").find("a").attr("href", "#");
+              });
+          } else {
+            $("#access_code_info").text("");
+          }
+        }).on("keydown", function(e) {
+          if (e.keyCode === 13) {
```

Comment:
Could you use a variable for the keycode? There's [a more descriptive API](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent#Browser_compatibility), but it's not widely supported :(